cpld_gridgen/man/man3/grdvars.3
.TH "grdvars" 3 "Wed Mar 13 2024" "Version 1.13.0" "cpld_gridgen" \" -*- nroff -*-
.ad l
.nh
.SH NAME
grdvars \- 
.SH SYNOPSIS
.br
.PP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "subroutine \fBallocate_all\fP"
.br
.RI "\fIAllocate grid variables\&. \fP"
.in -1c
.SS "Data Fields"

.in +1c
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBangle\fP"
.br
.RI "\fIThe rotation angle on Bu points\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBanglet\fP"
.br
.RI "\fIThe rotation angle on Ct points (opposite sense from angle) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBangq\fP"
.br
.RI "\fIThe grid rotation angle at the Bu (or corner) grid points of the super grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBareact\fP"
.br
.RI "\fIThe grid areas of the Ct grid cell in m2\&. \fP"
.ti -1c
.RI "logical \fBdebug\fP"
.br
.RI "\fIflag indicating whether grid information should be printed for debugging purposes Default is false \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBdlatbu\fP"
.br
.RI "\fIThe latitude spacing between Bu points at the grid bottom\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBdlatcv\fP"
.br
.RI "\fIThe latitude spacing between Cv points at the grid bottom\&. \fP"
.ti -1c
.RI "logical \fBdo_postwgts\fP"
.br
.RI "\fIflag indicating whether then ESMF weights to regrid from the tripole grid to a rectilinear grid should be generated\&. \fP"
.ti -1c
.RI "real(real_kind), dimension(:,:), 
.br
allocatable \fBdp4\fP"
.br
.RI "\fIThe ocean depth from a MOM6 topog file, stored as real*4 (m) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBdp8\fP"
.br
.RI "\fIThe ocean depth from a MOM6 topog file, stored as real*8 (m) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBdx\fP"
.br
.RI "\fIThe grid cell width in meters of the supergrid in the x-direction (i-dimension) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBdy\fP"
.br
.RI "\fIThe grid cell width in meters of the supergrid in the y-direction (j-dimension) \fP"
.ti -1c
.RI "logical \fBeditmask\fP"
.br
.RI "\fIflag indicating whether the MOM6 land mask should be edited\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBhte\fP"
.br
.RI "\fIThe grid cell width in centimeters of the CICE6 grid in the y-direction (j-dimension) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBhtn\fP"
.br
.RI "\fIThe grid cell width in centimeters of the CICE6 grid in the x-direction (i-dimension) \fP"
.ti -1c
.RI "integer(int_kind), dimension(2) \fBipole\fP"
.br
.RI "\fIthe i-index for both pole locations along the top-most row \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBivertbu\fP"
.br
.RI "\fIThe i-offsets of the Ct grid at each Bu(i,j) which determine the 4 vertices of each Bu grid point in i\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP), parameter \fBivertct\fP = (/0, -1, -1, 0/)"
.br
.RI "\fIThe i-offsets of the Bu grid at each Ct(i,j) which determine the 4 vertices of each Ct grid grid point in i\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBivertcu\fP"
.br
.RI "\fIThe i-offsets of the Cv grid at each Cu(i,j) which determine the 4 vertices of each Cu grid point in i\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBivertcv\fP"
.br
.RI "\fIThe i-offsets of the Cu grid at each Cv(i,j) which determine the 4 vertices of each Cv grid point in i\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBjvertbu\fP"
.br
.RI "\fIThe j-offsets of the Ct grid at each Bu(i,j) which determine the 4 vertices of each Bu grid point in j\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP), parameter \fBjvertct\fP = (/0, 0, -1, -1/)"
.br
.RI "\fIThe j-offsets of the Bu grid at each Ct(i,j) which determine the 4 vertices of each Ct grid point in j\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBjvertcu\fP"
.br
.RI "\fIThe j-offsets of the Cv grid at each Cu(i,j) which determine the 4 vertices of each Cu grid point in j\&. \fP"
.ti -1c
.RI "integer, dimension(\fBnv\fP) \fBjvertcv\fP"
.br
.RI "\fIThe j-offsets of the Cu grid at each Cv(i,j) which determine the 4 vertices of each Cv grid point in j\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlatbu\fP"
.br
.RI "\fIThe latitude of the corner points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlatbu_vert\fP"
.br
.RI "\fIThe latitudes of the 4 vertices of each Bu grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlatct\fP"
.br
.RI "\fIThe latitude of the center (tracer) grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlatct_vert\fP"
.br
.RI "\fIThe latitudes of the 4 vertices of each Ct grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlatcu\fP"
.br
.RI "\fIThe latitude of the u-velocity grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlatcu_vert\fP"
.br
.RI "\fIThe latitudes of the 4 vertices of each Cu grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlatcv\fP"
.br
.RI "\fIThe latitude of the v-velocity grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlatcv_vert\fP"
.br
.RI "\fIThe latitudes of the 4 vertices of each Cv grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlonbu\fP"
.br
.RI "\fIThe longitude of the corner points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlonbu_vert\fP"
.br
.RI "\fIThe longitudes of the 4 vertices of each Bu grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBlonct\fP"
.br
.RI "\fIThe longitude of the center (tracer) grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBlonct_vert\fP"
.br
.RI "\fIThe longitudes of the 4 vertices of each Ct grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBloncu\fP"
.br
.RI "\fIThe longitude of the u-velocity grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBloncu_vert\fP"
.br
.RI "\fIThe longitudes of the 4 vertices of each Cu grid point\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBloncv\fP"
.br
.RI "\fIThe longitude of the v-velocity grid points on the C-grid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:,:), 
.br
allocatable \fBloncv_vert\fP"
.br
.RI "\fIThe longitudes of the 4 vertices of each Cv grid point\&. \fP"
.ti -1c
.RI "real(kind=real_kind), parameter \fBmasking_depth\fP = 0\&.0"
.br
.RI "\fIThe masking depth for MOM6\&. \fP"
.ti -1c
.RI "real(kind=real_kind), parameter \fBmaximum_depth\fP = 6500\&.0"
.br
.RI "\fIThe maximum depth for MOM6\&. \fP"
.ti -1c
.RI "real(kind=real_kind), parameter \fBmaximum_lat\fP = 88\&.0"
.br
.RI "\fIThe maximum latitude for water points for WW3\&. \fP"
.ti -1c
.RI "real(kind=real_kind), parameter \fBminimum_depth\fP = 9\&.5"
.br
.RI "\fIThe minimum depth for MOM6\&. \fP"
.ti -1c
.RI "integer, parameter \fBncoord\fP = 2*4\&."
.br
.RI "\fIthe number of coord pairs (lat,lon) for each of 4 stagger locations \fP"
.ti -1c
.RI "integer \fBni\fP"
.br
.RI "\fIi-dimension of output grid \fP"
.ti -1c
.RI "integer \fBnj\fP"
.br
.RI "\fIj-dimension of output grid \fP"
.ti -1c
.RI "integer \fBnpx\fP"
.br
.RI "\fIi or j-dimension of fv3 tile \fP"
.ti -1c
.RI "integer, parameter \fBnv\fP = 4\&."
.br
.RI "\fIthe number of vertices for each stagger location \fP"
.ti -1c
.RI "integer, parameter \fBnvars\fP = \fBncoord\fP + \fBnverts\fP"
.br
.RI "\fIthe total number of cooridinate variables \fP"
.ti -1c
.RI "integer, parameter \fBnverts\fP = 2*4\&."
.br
.RI "\fIthe number of coord pairs (lat,lon) for the vertices of each stagger location \fP"
.ti -1c
.RI "integer \fBnx\fP"
.br
.RI "\fIi-dimension of MOM6 supergrid \fP"
.ti -1c
.RI "integer \fBny\fP"
.br
.RI "\fIj-dimension of MOM6 supergrid \fP"
.ti -1c
.RI "logical \fBroottask\fP"
.br
.RI "\fIflag indicating whether this is the roottask \fP"
.ti -1c
.RI "real(dbl_kind) \fBsg_maxlat\fP"
.br
.RI "\fIthe maximum latitute present in the supergrid file \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBulat\fP"
.br
.RI "\fIThe latitude points (on the Bu grid) for CICE6 (radians) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBulon\fP"
.br
.RI "\fIThe longitude points (on the Bu grid) for CICE6 (radians) \fP"
.ti -1c
.RI "real(real_kind), dimension(:,:), 
.br
allocatable \fBwet4\fP"
.br
.RI "\fIThe ocean mask from a MOM6 mask file, stored as real*4 (nd) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBwet8\fP"
.br
.RI "\fIThe ocean mask from a MOM6 mask file, stored as real*8 (nd) \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBx\fP"
.br
.RI "\fIThe longitudes of the MOM6 supergrid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBxlatct\fP"
.br
.RI "\fIThe latitude of the Ct grid points on the opposite side of the tripole seam\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBxlatcu\fP"
.br
.RI "\fIThe latitude of the Cu grid points on the opposite side of the tripole seam\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBxlonct\fP"
.br
.RI "\fIThe longitude of the Ct grid points on the opposite side of the tripole seam\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:), 
.br
allocatable \fBxloncu\fP"
.br
.RI "\fIThe longitude of the Cu grid points on the opposite side of the tripole seam\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBxsgp1\fP"
.br
.RI "\fIThe longitudes of the super-grid replicated across the tripole seam\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBy\fP"
.br
.RI "\fIThe latitudes of the MOM6 supergrid\&. \fP"
.ti -1c
.RI "real(dbl_kind), dimension(:,:), 
.br
allocatable \fBysgp1\fP"
.br
.RI "\fIThe latitudes of the super-grid replicated across the tripole seam\&. \fP"
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 8 of file grdvars\&.F90\&.
.SH "Member Function/Subroutine Documentation"
.PP 
.SS "subroutine grdvars::allocate_all ()"

.PP
Allocate grid variables\&. 
.PP
\fBAuthor:\fP
.RS 4
Denise Worthen 
.RE
.PP

.PP
Definition at line 173 of file grdvars\&.F90\&.
.PP
Referenced by gen_fixgrid()\&.
.SH "Field Documentation"
.PP 
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::angle"

.PP
The rotation angle on Bu points\&. 
.PP
Definition at line 105 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::anglet"

.PP
The rotation angle on Ct points (opposite sense from angle) 
.PP
Definition at line 103 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::angq"

.PP
The grid rotation angle at the Bu (or corner) grid points of the super grid\&. 
.PP
Definition at line 71 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::areact"

.PP
The grid areas of the Ct grid cell in m2\&. 
.PP
Definition at line 102 of file grdvars\&.F90\&.
.SS "logical grdvars::debug"

.PP
flag indicating whether grid information should be printed for debugging purposes Default is false 
.PP
Definition at line 23 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:), allocatable grdvars::dlatbu"

.PP
The latitude spacing between Bu points at the grid bottom\&. 
.PP
Definition at line 136 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:), allocatable grdvars::dlatcv"

.PP
The latitude spacing between Cv points at the grid bottom\&. 
.PP
Definition at line 138 of file grdvars\&.F90\&.
.SS "logical grdvars::do_postwgts"

.PP
flag indicating whether then ESMF weights to regrid from the tripole grid to a rectilinear grid should be generated\&. Default is false\&. 
.PP
Definition at line 26 of file grdvars\&.F90\&.
.SS "real(real_kind), dimension(:,:), allocatable grdvars::dp4"

.PP
The ocean depth from a MOM6 topog file, stored as real*4 (m) 
.PP
Definition at line 146 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::dp8"

.PP
The ocean depth from a MOM6 topog file, stored as real*8 (m) 
.PP
Definition at line 148 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::dx"

.PP
The grid cell width in meters of the supergrid in the x-direction (i-dimension) 
.PP
Definition at line 74 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::dy"

.PP
The grid cell width in meters of the supergrid in the y-direction (j-dimension) 
.PP
Definition at line 76 of file grdvars\&.F90\&.
.SS "logical grdvars::editmask"

.PP
flag indicating whether the MOM6 land mask should be edited\&. Default is false 
.PP
Definition at line 21 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::hte"

.PP
The grid cell width in centimeters of the CICE6 grid in the y-direction (j-dimension) 
.PP
Definition at line 158 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::htn"

.PP
The grid cell width in centimeters of the CICE6 grid in the x-direction (i-dimension) 
.PP
Definition at line 156 of file grdvars\&.F90\&.
.SS "integer(int_kind), dimension(2) grdvars::ipole"

.PP
the i-index for both pole locations along the top-most row 
.PP
Definition at line 41 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::ivertbu"

.PP
The i-offsets of the Ct grid at each Bu(i,j) which determine the 4 vertices of each Bu grid point in i\&. 
.PP
Definition at line 62 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP), parameter grdvars::ivertct = (/0, -1, -1, 0/)"

.PP
The i-offsets of the Bu grid at each Ct(i,j) which determine the 4 vertices of each Ct grid grid point in i\&. 
.PP
Definition at line 44 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::ivertcu"

.PP
The i-offsets of the Cv grid at each Cu(i,j) which determine the 4 vertices of each Cu grid point in i\&. 
.PP
Definition at line 56 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::ivertcv"

.PP
The i-offsets of the Cu grid at each Cv(i,j) which determine the 4 vertices of each Cv grid point in i\&. 
.PP
Definition at line 50 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::jvertbu"

.PP
The j-offsets of the Ct grid at each Bu(i,j) which determine the 4 vertices of each Bu grid point in j\&. 
.PP
Definition at line 65 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP), parameter grdvars::jvertct = (/0, 0, -1, -1/)"

.PP
The j-offsets of the Bu grid at each Ct(i,j) which determine the 4 vertices of each Ct grid point in j\&. 
.PP
Definition at line 47 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::jvertcu"

.PP
The j-offsets of the Cv grid at each Cu(i,j) which determine the 4 vertices of each Cu grid point in j\&. 
.PP
Definition at line 59 of file grdvars\&.F90\&.
.SS "integer, dimension(\fBnv\fP) grdvars::jvertcv"

.PP
The j-offsets of the Cu grid at each Cv(i,j) which determine the 4 vertices of each Cv grid point in j\&. 
.PP
Definition at line 53 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::latbu"

.PP
The latitude of the corner points on the C-grid\&. These are equivalent to u,v velocity grid points on the B-grid 
.PP
Definition at line 96 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:,:), allocatable grdvars::latbu_vert"

.PP
The latitudes of the 4 vertices of each Bu grid point\&. 
.PP
Definition at line 122 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::latct"

.PP
The latitude of the center (tracer) grid points on the C-grid\&. 
.PP
Definition at line 84 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:,:), allocatable grdvars::latct_vert"

.PP
The latitudes of the 4 vertices of each Ct grid point\&. 
.PP
Definition at line 107 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::latcu"

.PP
The latitude of the u-velocity grid points on the C-grid\&. 
.PP
Definition at line 92 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:,:), allocatable grdvars::latcu_vert"

.PP
The latitudes of the 4 vertices of each Cu grid point\&. 
.PP
Definition at line 117 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::latcv"

.PP
The latitude of the v-velocity grid points on the C-grid\&. 
.PP
Definition at line 88 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:,:), allocatable grdvars::latcv_vert"

.PP
The latitudes of the 4 vertices of each Cv grid point\&. 
.PP
Definition at line 112 of file grdvars\&.F90\&.
.SS "real(dbl_kind), dimension(:,:), allocatable grdvars::lonbu"

.PP

sorc/cpld_gridgen.fd/grdvars.F90
!> @file
!! @brief Define and allocate required grid variables
!! @author Denise.Worthen@noaa.gov
!!
!> This module contains the grid variables
!! @author Denise.Worthen@noaa.gov

module grdvars

  use gengrid_kinds, only : dbl_kind, real_kind, int_kind

  implicit none

  integer :: ni                                                    !< i-dimension of output grid
  integer :: nj                                                    !< j-dimension of output grid
  integer :: npx                                                   !< i or j-dimension of fv3 tile

  integer :: nx                                                    !< i-dimension of MOM6 supergrid
  integer :: ny                                                    !< j-dimension of MOM6 supergrid

  logical :: editmask                                              !< flag indicating whether the MOM6 land mask
                                                                   !! should be edited. Default is false
  logical :: debug                                                 !< flag indicating whether grid information
                                                                   !! should be printed for debugging purposes
                                                                   !! Default is false
  logical :: do_postwgts                                           !< flag indicating whether then ESMF weights to
                                                                   !! regrid from the tripole grid to a rectilinear
                                                                   !! grid should be generated. Default is false.
  logical :: roottask                                              !< flag indicating whether this is the roottask

  integer, parameter :: nv = 4.                                    !< the number of vertices for each stagger location
  integer, parameter :: ncoord = 2*4.                              !< the number of coord pairs (lat,lon) for each of
                                                                   !! 4 stagger locations
  integer, parameter :: nverts = 2*4.                              !< the number of coord pairs (lat,lon) for the
                                                                   !! vertices of each stagger location
  integer, parameter ::  nvars = ncoord + nverts                   !< the total number of cooridinate variables


  real(dbl_kind)     :: sg_maxlat                                  !< the maximum latitute present in the supergrid
                                                                   !! file
  integer(int_kind)  :: ipole(2)                                   !< the i-index for both pole locations
                                                                   !! along the top-most row

  integer, parameter, dimension(nv) :: iVertCt = (/0, -1, -1,  0/) !< The i-offsets of the Bu grid at each Ct(i,j)
                                                                   !! which determine the 4 vertices of each Ct grid
                                                                   !! grid point in i
  integer, parameter, dimension(nv) :: jVertCt = (/0,  0, -1, -1/) !< The j-offsets of the Bu grid at each Ct(i,j)
                                                                   !! which determine the 4 vertices of each Ct
                                                                   !! grid point in j
  integer, dimension(nv) :: iVertCv                                !< The i-offsets of the Cu grid at each Cv(i,j)
                                                                   !! which determine the 4 vertices of each Cv
                                                                   !! grid point in i
  integer, dimension(nv) :: jVertCv                                !< The j-offsets of the Cu grid at each Cv(i,j)
                                                                   !! which determine the 4 vertices of each Cv
                                                                   !! grid point in j
  integer, dimension(nv) :: iVertCu                                !< The i-offsets of the Cv grid at each Cu(i,j)
                                                                   !! which determine the 4 vertices of each Cu
                                                                   !! grid point in i
  integer, dimension(nv) :: jVertCu                                !< The j-offsets of the Cv grid at each Cu(i,j)
                                                                   !! which determine the 4 vertices of each Cu
                                                                   !! grid point in j
  integer, dimension(nv) :: iVertBu                                !< The i-offsets of the Ct grid at each Bu(i,j)
                                                                   !! which determine the 4 vertices of each Bu
                                                                   !! grid point in i
  integer, dimension(nv) :: jVertBu                                !< The j-offsets of the Ct grid at each Bu(i,j)
                                                                   !! which determine the 4 vertices of each Bu
                                                                   !! grid point in j
  ! Super-grid source grid variables
  real(dbl_kind), allocatable, dimension(:,:)   :: x               !< The longitudes of the MOM6 supergrid
  real(dbl_kind), allocatable, dimension(:,:)   :: y               !< The latitudes of the MOM6 supergrid
  real(dbl_kind), allocatable, dimension(:,:)   :: angq            !< The grid rotation angle at the Bu (or corner)
                                                                   !! grid points of the super grid

  real(dbl_kind), allocatable, dimension(:,:)   :: dx              !< The grid cell width in meters of the supergrid
                                                                   !! in the x-direction (i-dimension)
  real(dbl_kind), allocatable, dimension(:,:)   :: dy              !< The grid cell width in meters of the supergrid
                                                                   !! in the y-direction (j-dimension)
  real(dbl_kind), allocatable, dimension(:,:) :: xsgp1             !< The longitudes of the super-grid replicated
                                                                   !! across the tripole seam
  real(dbl_kind), allocatable, dimension(:,:) :: ysgp1             !< The latitudes of the super-grid replicated
                                                                   !! across the tripole seam

  ! Output grid variables
  real(dbl_kind), allocatable, dimension(:,:) :: latCt             !< The latitude of the center (tracer) grid points
                                                                   !! on the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: lonCt             !< The longitude of the center (tracer) grid
                                                                   !! points on the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: latCv             !< The latitude of the v-velocity grid points on
                                                                   !! the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: lonCv             !< The longitude of the v-velocity grid points on
                                                                   !! the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: latCu             !< The latitude of the u-velocity grid points on
                                                                   !! the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: lonCu             !< The longitude of the u-velocity grid points on
                                                                   !! the C-grid
  real(dbl_kind), allocatable, dimension(:,:) :: latBu             !< The latitude of the corner points on the C-grid.
                                                                   !! These are equivalent to u,v velocity grid
                                                                   !! points on the B-grid
  real(dbl_kind), allocatable, dimension(:,:) :: lonBu             !< The longitude of the corner points on the
                                                                   !! C-grid. These are equivalent to u,v velocity
                                                                   !! grid points on the B-grid
  real(dbl_kind), allocatable, dimension(:,:) :: areaCt            !< The grid areas of the Ct grid cell in m2
  real(dbl_kind), allocatable, dimension(:,:) :: anglet            !< The rotation angle on Ct points (opposite sense
                                                                   !! from angle)
  real(dbl_kind), allocatable, dimension(:,:) :: angle             !< The rotation angle on Bu points
  real(dbl_kind), allocatable, dimension(:,:) :: angchk            !< The rotation angle on Ct points, as calculated by
                                                                   !! CICE internally using angle on Bu

  real(dbl_kind), allocatable, dimension(:,:,:) :: latCt_vert      !< The latitudes of the 4 vertices of each Ct grid
                                                                   !! point
  real(dbl_kind), allocatable, dimension(:,:,:) :: lonCt_vert      !< The longitudes of the 4 vertices of each Ct
                                                                   !! grid point

  real(dbl_kind), allocatable, dimension(:,:,:) :: latCv_vert      !< The latitudes of the 4 vertices of each Cv grid
                                                                   !! point
  real(dbl_kind), allocatable, dimension(:,:,:) :: lonCv_vert      !< The longitudes of the 4 vertices of each Cv
                                                                   !! grid point

  real(dbl_kind), allocatable, dimension(:,:,:) :: latCu_vert      !< The latitudes of the 4 vertices of each Cu grid
                                                                   !! point
  real(dbl_kind), allocatable, dimension(:,:,:) :: lonCu_vert      !< The longitudes of the 4 vertices of each Cu
                                                                   !! grid point

  real(dbl_kind), allocatable, dimension(:,:,:) :: latBu_vert      !< The latitudes of the 4 vertices of each Bu grid
                                                                   !! point
  real(dbl_kind), allocatable, dimension(:,:,:) :: lonBu_vert      !< The longitudes of the 4 vertices of each Bu
                                                                   !! grid point


  real(dbl_kind), allocatable, dimension(:) :: xlonCt              !< The longitude of the Ct grid points on the
                                                                   !! opposite side of the tripole seam
  real(dbl_kind), allocatable, dimension(:) :: xlatCt              !< The latitude of the Ct grid points on the
                                                                   !! opposite side of the tripole seam
  real(dbl_kind), allocatable, dimension(:) :: xangCt              !< The rotation angle on the Ct grid points on the
                                                                   !! opposite side of the tripole seam

  real(dbl_kind), allocatable, dimension(:) :: xlonCu              !< The longitude of the Cu grid points on the
                                                                   !! opposite side of the tripole seam
  real(dbl_kind), allocatable, dimension(:) :: xlatCu              !< The latitude of the Cu grid points on the
                                                                   !! opposite side of the tripole seam
  real(dbl_kind), allocatable, dimension(:) :: dlatBu              !< The latitude spacing between Bu points at the
                                                                   !! grid bottom
  real(dbl_kind), allocatable, dimension(:) :: dlatCv              !< The latitude spacing between Cv points at the
                                                                   !! grid bottom
  ! MOM6 fix fields
  real(real_kind), allocatable, dimension(:,:) :: wet4             !< The ocean mask from a MOM6 mask file, stored as
                                                                   !! real*4 (nd)
  real(dbl_kind),  allocatable, dimension(:,:) :: wet8             !< The ocean mask from a MOM6 mask file, stored as
                                                                   !! real*8 (nd)

  real(real_kind), allocatable, dimension(:,:) :: dp4              !< The ocean depth from a MOM6 topog file, stored
                                                                   !! as real*4 (m)
  real(dbl_kind),  allocatable, dimension(:,:) :: dp8              !< The ocean depth from a MOM6 topog file, stored
                                                                   !! as real*8 (m)

  ! CICE6 fields
  real(dbl_kind), allocatable, dimension(:,:) :: ulon              !< The longitude points (on the Bu grid) for CICE6
                                                                   !! (radians)
  real(dbl_kind), allocatable, dimension(:,:) :: ulat              !< The latitude points (on the Bu grid) for CICE6
                                                                   !! (radians)
  real(dbl_kind), allocatable, dimension(:,:) ::  htn              !< The grid cell width in centimeters of the CICE6
                                                                   !! grid in the x-direction (i-dimension)
  real(dbl_kind), allocatable, dimension(:,:) ::  hte              !< The grid cell width in centimeters of the CICE6
                                                                   !! grid in the y-direction (j-dimension)

  real(kind=real_kind), parameter :: minimum_depth = 9.5           !< The minimum depth for MOM6
  real(kind=real_kind), parameter :: maximum_depth = 6500.0        !< The maximum depth for MOM6
  real(kind=real_kind), parameter :: masking_depth = 0.0           !< The masking depth for MOM6. Depths shallower than
                                                                   !! minimum_depth but deeper than masking_depth are
                                                                   !! rounded to minimum_depth
  real(kind=real_kind), parameter :: maximum_lat = 88.0            !< The maximum latitude for water points for WW3

contains
  !> Allocate grid variables
  !!
  !! @author Denise Worthen

  subroutine allocate_all

    allocate( x(0:nx,0:ny),  y(0:nx,0:ny), angq(0:nx,0:ny) )
    allocate(  dx(nx,0:ny), dy(0:nx,ny) )

    allocate( xsgp1(0:nx,0:ny+1), ysgp1(0:nx,0:ny+1) )

    allocate( latCt(ni,nj), lonCt(ni,nj) )
    allocate( latCv(ni,nj), lonCv(ni,nj) )
    allocate( latCu(ni,nj), lonCu(ni,nj) )
    allocate( latBu(ni,nj), lonBu(ni,nj) )

    allocate( areaCt(ni,nj), anglet(ni,nj), angle(ni,nj), angchk(ni,nj))

    allocate( latCt_vert(ni,nj,nv), lonCt_vert(ni,nj,nv) )
    allocate( latCv_vert(ni,nj,nv), lonCv_vert(ni,nj,nv) )
    allocate( latCu_vert(ni,nj,nv), lonCu_vert(ni,nj,nv) )
    allocate( latBu_vert(ni,nj,nv), lonBu_vert(ni,nj,nv) )

    allocate( xlonCt(ni), xlatCt(ni), xangCt(ni) )
    allocate( xlonCu(ni), xlatCu(ni) )
    allocate( dlatBu(ni), dlatCv(ni) )

    allocate( wet4(ni,nj) )
    allocate( wet8(ni,nj) )

    allocate(  dp4(ni,nj) )
    allocate(  dp8(ni,nj) )

    allocate( ulon(ni,nj), ulat(ni,nj) )
    allocate(  htn(ni,nj),  hte(ni,nj) )

  end subroutine allocate_all

end module grdvars
